lib/no_brainer/criteria/where.rb

```diff
@@ -285,11 +285,22 @@ def parse_multi_value(op, value, options={})
   def parse_clause_stub_eq(key, value)
     case value
     when Range  then instantiate_binary_op(key, :between, value)
-    when Regexp then instantiate_binary_op(key, :match, value.inspect[1..-2])
+    when Regexp then instantiate_binary_op(key, :match, translate_regexp_to_re2_syntax(value))
     else instantiate_binary_op(key, :eq, value)
     end
   end
 
+  def translate_regexp_to_re2_syntax(value)
+    options = ""
+    options << "i" if 0 != value.options & Regexp::IGNORECASE
+
+    if options.empty?
+      value.source
+    else
+      "(?#{options}:#{value.source})"
```

Comment:
From https://code.google.com/p/re2/wiki/Syntax it seems that we should rename `options` to `flags.

From the example on http://www.rethinkdb.com/api/ruby/match/ it seems that the parenthesis go around the flags, not the whole thing.

From the re2 docs:

> `(?flags)`	set flags within current group; non-capturing
> `(?flags:re)`	set flags during re; non-capturing

I don't understand the difference. Do you?